apps/api/src/controllers/applicationController.ts

```diff
@@ -252,3 +252,73 @@ export const submitRevision = async ({ applicationId }: { applicationId: number
 		return failure(message, error);
 	}
 };
+
+export const closeApplication = async ({
+	applicationId,
+	requesterId,
+	isDacMember = false,
+}: {
+	applicationId: number;
+	requesterId: string;
+	isDacMember?: boolean;
+}) => {
+	try {
+		const database = getDbInstance();
+		const service: ApplicationService = applicationSvc(database);
+		const result = await service.getApplicationById({ id: applicationId });
+
+		if (!result.success) {
+			return result;
+		}
+
+		const application = result.data;
+		const appStateManager = new ApplicationStateManager(application);
+
+		// Check if application is already closed
+		if (appStateManager.state === ApplicationStates.CLOSED) {
+			return failure('Application is already closed.', 'StateConflict');
+		}
+
+		// Authorization logic based on state
+		const isApplicant = application.user_id === requesterId;
+		let closeResult;
+
+		switch (appStateManager.state) {
+			case ApplicationStates.DRAFT:
+				if (!isApplicant) {
```

Comment:
These `!isApplicant` checks don't need to be repeated, you could move this right after line 283 and only do it once

Reply:
Hmm I see it's a little more complicated than that.

What I see is that `!isApplicant` causes an error no matter which state.
Only Dac Review has an extra case where it could be a Dac Member.

So I think the switch case could be condensed to two conditionals instead of 4 different cases.
something like:
`
const canCloseDac = !isDacMember && appStateManager.state === ApplicationStates.DAC_REVIEW;
if (!isApplicant) { 
    if (!canCloseDac) { return failure(...) } // if you need Dac specific messaging
    return failure(...)
}
`

And then the `appStateManager.closeDraft();` etc can be handled in their own block

Just see a lot of repeat error messages which could be reduced